profile/src/v100/app_preferences/gateways/radix_network.rs

```diff
@@ -0,0 +1,331 @@
+use crate::prelude::*;
+
+use crate::NetworkID::{self, *};
+
+/// A version of the Radix Network, for a NetworkID with an identifier (name) and display description (display name)
+#[derive(
+    Serialize,
+    Deserialize,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    Ord,
+    PartialOrd,
+    Hash,
+    derive_more::Display,
+    uniffi::Record,
+)]
+#[serde(rename_all = "camelCase")]
+#[display("{} ({})", self.display_description, self.id.discriminant())]
+pub struct RadixNetwork {
+    /// A String identifier (always lowercase) with the name of the Network that MUST match what Gateway returns.
+    #[serde(rename = "name")]
+    pub logical_name: String,
+
+    /// The canonical identifier of this network.
+    pub id: NetworkID,
+
+    /// A name of the network intended for display purposes only.
+    pub display_description: String,
+}
+
+impl Default for RadixNetwork {
+    fn default() -> Self {
+        Self::mainnet()
+    }
+}
+
+impl RadixNetwork {
+    fn declare(id: NetworkID, display: &str) -> Self {
+        Self {
+            logical_name: id.network_definition().logical_name,
+            id,
+            display_description: display.to_string(),
+        }
+    }
+}
+
+impl RadixNetwork {
+    /// The Radix mainnet, the "real" Network on which all launched Dapps and
+    /// assets with any real value resides.
+    pub fn mainnet() -> Self {
+        Self::declare(Mainnet, "Mainnet")
+    }
+
+    /// The primary public testnet of the Radix ecosystem, used by Dapp Developers
+    /// and RDX Works alike to test new features.
+    pub fn stokenet() -> Self {
+        Self::declare(Stokenet, "Stokenet")
+    }
+
+    /// A Betanet.
+    pub fn nebunet() -> Self {
+        Self::declare(Nebunet, "Betanet")
+    }
+
+    /// Was a Release Candidate for Babylon launch.
+    pub fn kisharnet() -> Self {
+        Self::declare(Kisharnet, "RCnet (Test Network)")
+    }
+
+    /// Was the second Release Candidate for Babylon launch.
+    pub fn ansharnet() -> Self {
+        Self::declare(Ansharnet, "RCnet-V2 (Test Network)")
+    }
+
+    /// Was the third Release Candidate for Babylon launch.
+    pub fn zabanet() -> Self {
+        Self::declare(Zabanet, "RCnet-V3 (Test Network)")
+    }
+
+    /// A testnet.
+    pub fn hammunet() -> Self {
+        Self::declare(Hammunet, "Hammunet (Test Network)")
+    }
+
+    /// A testnet.
+    pub fn enkinet() -> Self {
+        Self::declare(Enkinet, "Enkinet (Test Network)")
+    }
+
+    /// A testnet.
+    pub fn nergalnet() -> Self {
+        Self::declare(NetworkID::Nergalnet, "Nergalnet (Test Network)")
+    }
+
+    /// A testnet.
+    pub fn mardunet() -> Self {
+        Self::declare(NetworkID::Mardunet, "Mardunet (Test Network)")
+    }
+}
+
+impl HasPlaceholder for RadixNetwork {
+    fn placeholder() -> Self {
+        Self::mainnet()
+    }
+
+    fn placeholder_other() -> Self {
+        Self::stokenet()
+    }
+}
+
+impl RadixNetwork {
```

Comment:
this is logic, which perhaps ought be broken out of `v100` somehow? But as stated in description of this PR... there is not real cost of just copy pasting over the `v100` folder and renaming the copy `v101` and have these lookup logic duplicated... since the ones in `v100` will never be used once `v101` exists. 

If we were to try to break logic out, how to structure that code? Freestanding global functions? Or a `trait`? hmm